packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -41,33 +68,62 @@ export class CookieStorage implements BaseStorage {
  * or a customized storage. This class is mainly used for commerce-sdk-react library
  * to store authentication tokens.
  */
-export class LocalStorage implements BaseStorage {
-    constructor() {
+export class LocalStorage extends BaseStorage {
+    constructor(options?: BaseStorageOptions) {
+        super(options)
+
         if (typeof window === 'undefined') {
             throw new Error('LocalStorage is not avaliable on the current environment.')
         }
     }
     set(key: string, value: string) {
-        const oldValue = this.get(key)
-        window.localStorage.setItem(key, value)
+        const prefixedKey = this.getPrefixedKey(key)
+        const oldValue = this.get(prefixedKey)
+        window.localStorage.setItem(prefixedKey, value)
         const event = new StorageEvent('storage', {
-            key: key,
+            key: prefixedKey,
             oldValue: oldValue,
             newValue: value
         })
         window.dispatchEvent(event)
     }
     get(key: string) {
-        return window.localStorage.getItem(key) || ''
+        const prefixedKey = this.getPrefixedKey(key)
+        return window.localStorage.getItem(prefixedKey) || ''
     }
     delete(key: string) {
-        const oldValue = this.get(key)
-        window.localStorage.removeItem(key)
+        const prefixedKey = this.getPrefixedKey(key)
+        const oldValue = this.get(prefixedKey)
+        window.localStorage.removeItem(prefixedKey)
         const event = new StorageEvent('storage', {
-            key: key,
+            key: prefixedKey,
             oldValue: oldValue,
             newValue: null
         })
         window.dispatchEvent(event)
     }
 }
+
+const globalMap = new Map()
+
+export class MemoryStorage extends BaseStorage {
+    // private map = new Map<string, string>()
+    private map: Map<string, string>
+    constructor(options?: MemoryStorageOptions) {
+        super(options)
+
+        this.map = options?.sharedContext ? globalMap : new Map()
```

Comment:
Re: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/911#discussion_r1087152328

Our use case is to use a shared context. Is our use case common enough that it should be the default, or should we default to isolated contexts?

Reply:
I chose the default to be isolated maps, as I think it's the most natural. Is it weird that our only use of this is not the default.. maybe.. but I think that our use case of wanting global storage from lambda run to another is a little wweird.